### PR TITLE
Update to v3.3

### LINKED ISSUE #5: Ubuntu 20.04 and IoT Edge Runtime 1.4 - error while loading shared libraries libprotobuf.so.10 - No such file or directory
Hi,

We are using the eve-tools to be able to extract the endorsement key from the TPM through EVE-OS to be able to do DPS provisioning on Azure.

Everything was working well in Ubuntu 18.04. But since, I upgraded the VM to Ubuntu 20.04, it's not working properly.
Here are some logs returned when typing the command `sudo aziotctl system logs`

```
Apr 11 06:12:53 M6ZZ2553 aziot-edged[92729]: 2023-04-11T06:12:53Z [WARN] - Failed to send HTTP request (attempt 2 of 2): connection error: Connection reset by peer (os error 104)
Apr 11 06:12:53 M6ZZ2553 aziot-edged[92729]: 2023-04-11T06:12:53Z [WARN] - Failed to reprovision: connection error: Connection reset by peer (os error 104)
Apr 11 06:12:53 M6ZZ2553 systemd[1]: Stopped Azure IoT Keys Service.
Apr 11 06:12:53 M6ZZ2553 systemd[1]: Started Azure IoT Identity Service.
Apr 11 06:12:53 M6ZZ2553 aziot-identityd[93198]: /usr/libexec/aziot-identity-service/aziot-identityd: error while loading shared libraries: libprotobuf.so.10: cannot open shared object file: No such file or directory
Apr 11 06:12:53 M6ZZ2553 systemd[1]: Started Azure IoT Keys Service.
Apr 11 06:12:53 M6ZZ2553 systemd[1]: aziot-identityd.service: Main process exited, code=exited, status=127/n/a
Apr 11 06:12:53 M6ZZ2553 systemd[1]: aziot-identityd.service: Failed with result 'exit-code'.
Apr 11 06:12:53 M6ZZ2553 aziot-keyd[93201]: /usr/libexec/aziot-identity-service/aziot-keyd: error while loading shared libraries: libprotobuf.so.10: cannot open shared object file: No such file or directory
Apr 11 06:12:53 M6ZZ2553 systemd[1]: aziot-keyd.service: Main process exited, code=exited, status=127/n/a
Apr 11 06:12:53 M6ZZ2553 systemd[1]: aziot-keyd.service: Failed with result 'exit-code'.
Apr 11 06:12:58 M6ZZ2553 aziot-edged[92729]: 2023-04-11T06:12:58Z [INFO] - Obtaining Edge device provisioning data...
Apr 11 06:12:58 M6ZZ2553 systemd[1]: aziot-identityd.service: Scheduled restart job, restart counter is at 6186.
Apr 11 06:12:58 M6ZZ2553 systemd[1]: aziot-keyd.service: Scheduled restart job, restart counter is at 669.
Apr 11 06:12:58 M6ZZ2553 aziot-edged[92729]: 2023-04-11T06:12:58Z [WARN] - Failed to send HTTP request (attempt 1 of 2): connection error: Connection reset by peer (os error 104)
Apr 11 06:12:58 M6ZZ2553 systemd[1]: Stopped Azure IoT Identity Service.
Apr 11 06:12:58 M6ZZ2553 systemd[1]: Stopped Azure IoT Keys Service.
Apr 11 06:12:58 M6ZZ2553 systemd[1]: Started Azure IoT Identity Service.
Apr 11 06:12:58 M6ZZ2553 systemd[1]: Started Azure IoT Keys Service.
Apr 11 06:12:58 M6ZZ2553 systemd[1]: aziot-identityd.service: Main process exited, code=exited, status=127/n/a
Apr 11 06:12:58 M6ZZ2553 aziot-identityd[93221]: /usr/libexec/aziot-identity-service/aziot-identityd: error while loading shared libraries: libprotobuf.so.10: cannot open shared object file: No such file or directory
Apr 11 06:12:58 M6ZZ2553 aziot-keyd[93222]: /usr/libexec/aziot-identity-service/aziot-keyd: error while loading shared libraries: libprotobuf.so.10: cannot open shared object file: No such file or directory
Apr 11 06:12:58 M6ZZ2553 systemd[1]: aziot-identityd.service: Failed with result 'exit-code'.
Apr 11 06:12:58 M6ZZ2553 systemd[1]: aziot-keyd.service: Main process exited, code=exited, status=127/n/a
Apr 11 06:12:58 M6ZZ2553 systemd[1]: aziot-keyd.service: Failed with result 'exit-code'.
Apr 11 06:13:03 M6ZZ2553 systemd[1]: aziot-identityd.service: Scheduled restart job, restart counter is at 6187.
Apr 11 06:13:03 M6ZZ2553 aziot-edged[92729]: 2023-04-11T06:13:03Z [WARN] - Failed to send HTTP request (attempt 2 of 2): connection error: Connection reset by peer (os error 104)
Apr 11 06:13:03 M6ZZ2553 aziot-edged[92729]: 2023-04-11T06:13:03Z [ERR!] - Failed to obtain device identity: connection error: Connection reset by peer (os error 104)
Apr 11 06:13:03 M6ZZ2553 aziot-edged[92729]: 2023-04-11T06:13:03Z [INFO] - Requesting device reprovision
Apr 11 06:13:03 M6ZZ2553 aziot-edged[92729]: 2023-04-11T06:13:03Z [WARN] - Failed to clear provisioning cache before reprovision: No such file or directory (os error 2)
Apr 11 06:13:03 M6ZZ2553 aziot-edged[92729]: 2023-04-11T06:13:03Z [WARN] - Failed to send HTTP request (attempt 1 of 2): error trying to connect: Connection refused (os error 111)
Apr 11 06:13:03 M6ZZ2553 systemd[1]: aziot-keyd.service: Scheduled restart job, restart counter is at 670.
Apr 11 06:13:03 M6ZZ2553 systemd[1]: Stopped Azure IoT Identity Service.
Apr 11 06:13:03 M6ZZ2553 systemd[1]: Stopped Azure IoT Keys Service.
Apr 11 06:13:03 M6ZZ2553 aziot-identityd[93240]: /usr/libexec/aziot-identity-service/aziot-identityd: error while loading shared libraries: libprotobuf.so.10: cannot open shared object file: No such file or directory
Apr 11 06:13:03 M6ZZ2553 aziot-keyd[93241]: /usr/libexec/aziot-identity-service/aziot-keyd: error while loading shared libraries: libprotobuf.so.10: cannot open shared object file: No such file or directory
Apr 11 06:13:03 M6ZZ2553 systemd[1]: Started Azure IoT Identity Service.
Apr 11 06:13:03 M6ZZ2553 systemd[1]: Started Azure IoT Keys Service.
Apr 11 06:13:03 M6ZZ2553 systemd[1]: aziot-identityd.service: Main process exited, code=exited, status=127/n/a
Apr 11 06:13:03 M6ZZ2553 systemd[1]: aziot-identityd.service: Failed with result 'exit-code'.
Apr 11 06:13:03 M6ZZ2553 systemd[1]: aziot-keyd.service: Main process exited, code=exited, status=127/n/a
Apr 11 06:13:03 M6ZZ2553 systemd[1]: aziot-keyd.service: Failed with result 'exit-code'.
```

**1.) Do you have any insights? Is just a missing dependency?**

_2.) Also, I guess we are not using the libiothsm package anymore, so I guess some step of the EVE-Tools that was overriding the libiothsm are not needed anymore? https://learn.microsoft.com/en-us/azure/iot-edge/how-to-update-iot-edge?view=iotedge-1.4&tabs=linux#special-case-update-from-10-or-11-to-latest-release_

### PR DESCRIPTION
**Pull request by shjala**
This pull request incorporates several updates and improvements to the project. Here's a summary of the changes:

- Add readme on how to run tests
- Updated install documents to reflect latest changes
- Added the required file and updated Dockerfile to build azure-on-eve for versions 1.2 and 1.4 and make a debian package for easy installation
- minor changes to keep the tpm-import command backward compatible
- fixes https://github.com/lf-edge/eve-tools/issues/5

### CODE REVIEW

**Comment by shjala** [comment]
@eriknordmark @cshari-zededa ping.

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,52 @@
 # Copyright (c) 2019 Zededa, Inc.
 # SPDX-License-Identifier: Apache-2.0
-FROM debian:9.11
-RUN apt update && \ 
-    apt install -y g++ make libprotobuf-dev \
-                   libprotoc-dev protobuf-compiler \
-                   cmake libssl-dev libcurl4-openssl-dev uuid-dev
+FROM ubuntu:20.04 as build
+ARG IOTD_COMMIT_ID=15f59c8bd33b1fd8581a74ae6e5ea145c8cb1b9b
+ARG DEBIAN_FRONTEND=noninteractive
```
Comment (eriknordmark) [inline Dockerfile]: Where is this arg used?

File: INSTALL.md [hunk 2]
```diff
@@ -44,6 +48,22 @@ sudo cp libiothsm.so* /usr/lib
...
+git clone https://github.com/Azure/iot-identity-service.git
+cp iot-identity-service.diff iot-identity-service/
+cd iot-identity-service
+git reset --hard 15f59c8bd33b1fd8581a74ae6e5ea145c8cb1b9b
+git apply iot-identity-service.diff
+FORCE_NO_UNITTEST=1 make
+```
+
 
 #### 5. After this restart IoT Edge Runtime, for the plugin to take effect:
```
Comment (eriknordmark) [inline INSTALL.md]: Does markdown automatically set the section number? If not, this needs to be changed to 6.

**Comment by eriknordmark** [comment]
Hari is out this week - maybe @gkodali-zededa can review instead?

**Comment by eriknordmark** [comment]
@shjala does this by chance address the build issue in #5 ?

**Comment by shjala** [comment]
> @shjala does this by chance address the build issue in #5 ?

yes, it should.

File: Dockerfile (hunk 1 above)
Comment (shjala) [reply Dockerfile:5]: on line 38

Comment (shjala) [reply Dockerfile]: it is for apt-get, without this it might get stuck asking for keyboard layout or something.

File: INSTALL.md (hunk 2 above)
Comment (shjala) [reply INSTALL.md]: fixed

**Comment by eriknordmark** [comment]
> > @shjala does this by chance address the build issue in #5 ?
> 
> yes, it should.

If you add fixes #5 to the PR description then the issue will be automatically closed with a reference when this PR merges.

File: Dockerfile (hunk 1 above)
Comment (eriknordmark) [reply Dockerfile]: OK, so this becomes a setenv which is then passed to apt-get etc. It would be good to add a comment about that.

**Review by gkodali-zededa — COMMENTED** [review]
LGTM

**Pull request merged**